package/samplers/auto_sampler/_sampler.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from collections.abc import Sequence
+import threading
+from typing import Any
+from typing import TYPE_CHECKING
+
+from optuna.distributions import CategoricalDistribution
+from optuna.logging import get_logger
+from optuna.samplers import BaseSampler
+from optuna.samplers import CmaEsSampler
+from optuna.samplers import GPSampler
+from optuna.samplers import NSGAIISampler
+from optuna.samplers import RandomSampler
+from optuna.samplers import TPESampler
+from optuna.samplers._base import _process_constraints_after_trial
+from optuna.samplers._lazy_random_state import LazyRandomState
+from optuna.search_space import IntersectionSearchSpace
+from optuna.trial import TrialState
+
+
+if TYPE_CHECKING:
+    from optuna.distributions import BaseDistribution
+    from optuna.study import Study
+    from optuna.trial import FrozenTrial
+
+
+_MAXINT32 = (1 << 31) - 1
+_SAMPLER_KEY = "auto:sampler"
+# NOTE(nabenabe): The prefix `optuna.` enables us to use the optuna logger externally.
+_logger = get_logger(f"optuna.{__name__}")
+
+
+class _ThreadLocalSampler(threading.local):
+    _sampler: BaseSampler | None = None
```

Comment:
[nit] I'm sorry if I caused any confusion, but perhaps we can make these variables, class names, and fields public since we renamed the module from `auto_samper.sampler` to `auto_sampler._sampler`.

Please feel free to ignore this comment anyway.